lib/date.js

```diff
@@ -98,8 +98,17 @@ module.exports = (key, options) => {
         label: label === labelKey ? v.label : label
       });
     }));
+    const conditionalTranslate = (fieldKey) => {
+      let result = req.translate(fieldKey);
+      if (result === fieldKey) {
+        result = null;
+      }
+      return result;
+    };
```

Comment:
only a _minor_ thing - but I would move this function outside of the main export function - up where you see the other helpers. This is because here it has access to all of the internal variables such as key, fields etc... This could lead to some hard to find bugs if you rely on an outside variable etc. I guess this is why you renamed `key` to `fieldKey` (which doesn't really make sense) - if you move it outside of the module export function scope, you can go back to calling it `key` - which I prefer :)